pkg/provider/azure_loadbalancer.go

```diff
@@ -1123,11 +1159,19 @@ func (az *Cloud) reconcileIPSettings(pip *network.PublicIPAddress, service *v1.S
 	var changed bool
 
 	serviceName := getServiceName(service)
-	ipv6 := utilnet.IsIPv6String(service.Spec.ClusterIP)
-	if ipv6 {
+	isIPv6 := pip.PublicIPAddressVersion == network.IPv6
+	var clusterIP string
+	for _, ip := range service.Spec.ClusterIPs {
+		if (utilnet.IsIPv6String(ip) && isIPv6) || (!utilnet.IsIPv6String(ip) && !isIPv6) {
+			clusterIP = ip
+			break
+		}
+	}
+
+	if isIPv6 {
 		if !strings.EqualFold(string(pip.PublicIPAddressVersion), string(network.IPv6)) {
```

Comment:
It looks like a useless check.

Reply:
Thank you for pointing this out! I rechecked the code and it isn't useless because this methods is used to check if pip's IP version matches expected one.
However, there's still something to improve. I add a new `isIPv6 bool` parameter to this method so `clusterIP` is not used anymore. Also I added a UT for this method. PTAL.